cabal-install/Distribution/Client/CmdRun.hs

```diff
@@ -172,7 +172,7 @@ runAction (configFlags, configExFlags, installFlags, haddockFlags, testFlags)
 
     let
       scriptOrError script err = do
-        exists <- doesFileExist script
+        exists <- validScript script
```

Comment:
using the same `validScript` here has the consequence that

    cabal run myscript.hs

doesn't work anymore because it doesn't contain any path-seps; i.e. you'd have to say

    cabal run ./myscript.hs

now. The path-separator heuristic is only desirable for the shebang-heuristic